src/test/java/parser/JsonParserTest.java

```diff
@@ -0,0 +1,173 @@
+    void parserRewritesFileDuringWritingCartToFileWhichAlreadyExists() throws IOException {
+
+        Cart cart = new CartCreator().plainCart(NAME_PREFIX + "plainCart" + NAME_POSTFIX);
+
+        File file = new File(BASE_PATH + cart.getCartName() + ".json");
+        Assert.assertFalse(file.isFile());
+
+        String textInFile = "message";
+        new FileCreator().createFile(BASE_PATH, cart.getCartName() + ".json", textInFile);
+        Assert.assertTrue(file.isFile());
+
+        jsonParser.writeToFile(cart);
+        Assert.assertTrue(file.isFile());
+        String actualTextInFile = new MyFileReader().readFile(BASE_PATH, cart.getCartName() + ".json");
+
+        Assert.assertNotEquals(actualTextInFile, textInFile);
+    }
+
+    @Test(groups = {"integration"})
+    void writeNullCartToFile() {
+
+        Cart cart = null;
+
+        Assert.assertThrows(NullPointerException.class, () -> jsonParser.writeToFile(cart));
+    }
+
+    @Test(groups = {"integration"})
+    void writeEmptyCartToFile() throws IOException {
+        Cart cart = new CartCreator().emptyCart(NAME_PREFIX + "emptyCart" + NAME_POSTFIX);
+
+        jsonParser.writeToFile(cart);
+        String expectedText = new Gson().toJson(cart);
+
+        String actualText = new MyFileReader().readFile(BASE_PATH, cart.getCartName() + ".json");
+
+        Assert.assertEquals(expectedText, actualText);
+    }
+
+    @Test(groups = {"integration"})
+    void writePlainCartToFile() throws IOException {
+
+        Cart cart = new CartCreator().plainCart(NAME_PREFIX + "plainCart" + NAME_POSTFIX);
+
+        jsonParser.writeToFile(cart);
+        String expectedText = new Gson().toJson(cart);
+
+        String actualText = new MyFileReader().readFile(BASE_PATH, cart.getCartName() + ".json");
+
+        Assert.assertEquals(expectedText, actualText);
+    }
+
+    @Test(groups = {"integration"})
+    void writeBigCartToFile() throws IOException {
+
+        Cart cart = new CartCreator().bigCart(NAME_PREFIX + "bigCart" + NAME_POSTFIX);
+
+        jsonParser.writeToFile(cart);
+        String expectedText = new Gson().toJson(cart);
+
+        String actualText = new MyFileReader().readFile(BASE_PATH, cart.getCartName() + ".json");
+
+        Assert.assertEquals(expectedText, actualText);
+    }
+
+    @Test(groups = {"integration"})
+    void readFromFileNoFile() {
+
+        File file = new File("random_path");
+        Assert.assertThrows(NoSuchFileException.class, () -> jsonParser.readFromFile(file));
+    }
+
+    @Test(groups = {"integration"}, enabled = false)
+    void readFromFileWithUnexpectedStructure() throws IOException {
+
+        String fileName = NAME_PREFIX + "randomFileName" + NAME_POSTFIX + ".txt";
+        String text = "random text";
+        new FileCreator().createFile(BASE_PATH, fileName, text);
+
+        File file = new File(BASE_PATH + fileName);
+
+        Cart cart = jsonParser.readFromFile(file);
+
+        //I expect that in this case cart should be null.
```

Comment:
It is true. One more exception probably will be good to add.  But it is what we have now.
Excellent that you have test that catch it